Add a libInstPatch dependency

T-X opened the pull request:
libInstPatch is necessary to enable the DLS support in fluidsynth.

T-X committed:
Add a libInstPatch dependency for DLS support

libInstPatch is necessary to enable the DLS support in fluidsynth.

Signed-off-by: Linus Lüssing <linus.luessing@c0d3.blue>

T-X commented:
Note: The instructions added to .github/workflows/macos.yml here build fine on Linux / Debian Sid. Untested for Mac OS though. This would be to add Mac OS support to https://github.com/musescore/MuseScore/pull/28350, if it worked.

T-X commented:
I could at least compile this successfully through Github actions here: https://github.com/T-X/musescore_deps/actions/runs/15595044307

I needed to remove x86_64 from -DCMAKE_OSX_ARCHITECTURES though: https://github.com/T-X/musescore_deps/commit/6ab6bd9ab70f1ca4b33bd8b71acac691bba6d144. As the runner seems to be arm64 and had glib preinstalled. And seemed to cross-compile for x86_64 otherwise and would fail because of the architecture mismatch for glib:

```
...
ld: warning: ignoring file '/opt/homebrew/Cellar/glib/2.84.2/lib/libgobject-2.0.0.dylib': found architecture 'arm64', required architecture 'x86_64'
ld: warning: ignoring file '/opt/homebrew/Cellar/glib/2.84.2/lib/libgthread-2.0.0.dylib': found architecture 'arm64', required architecture 'x86_64'
ld: warning: ignoring file '/opt/homebrew/Cellar/glib/2.84.2/lib/libglib-2.0.0.dylib': found architecture 'arm64', required architecture 'x86_64'
ld: warning: ignoring file '/opt/homebrew/Cellar/gettext/0.25/lib/libintl.8.dylib': found architecture 'arm64', required architecture 'x86_64'
Undefined symbols for architecture x86_64:
  "_g_array_append_vals", referenced from:
      _convert_lookup_map_U in IpatchConverter.c.o
      _type_all_children in IpatchFile.c.o
...
```
https://github.com/T-X/musescore_deps/actions/runs/15593787240/job/43919124839#step:15:1971

And I had to add `-DCMAKE_POLICY_VERSION_MINIMUM=3.5` to most cmake calls in the workflow.

---

I'm not quite sure how I would verify that this PR would work with upstream MuseScore though as I don't seem to be able to enable MuseScore's actions in my fork.

cbjeukendrup commented:
We really do need x86_64, otherwise MuseScore won't run on Intel-based Macs anymore. That means we will need to compile a "universal" binary (i.e. arm64 and x86_64 in one binary) of Glib. 

> And I had to add `-DCMAKE_POLICY_VERSION_MINIMUM=3.5` to most cmake calls in the workflow.

That makes sense. These packages are all very old, and specify a very low `cmake_minimum_required`, which doesn't work anymore with recent CMake versions. As an alternative solution, you could check if newer package versions are available, that have a more recent `cmake_minimum_required`.

cbjeukendrup commented:
> I'm not quite sure how I would verify that this PR would work with upstream MuseScore though as I don't seem to be able to enable MuseScore's actions in my fork.

That should be possible. If you go to your fork's page, then to Actions, and then click a specific action in the sidebar, you should be able to enable it.

T-X commented:
> > I'm not quite sure how I would verify that this PR would work with upstream MuseScore though as I don't seem to be able to enable MuseScore's actions in my fork.
> 
> That should be possible. If you go to your fork's page, then to Actions, and then click a specific action in the sidebar, you should be able to enable it.

I've updated the setup.sh to wget from my musescore_deps fork, then clicked on "Build: macs OS"->"Run workflow"->Branch: "pr-fluidsynth-dls-support-macos" but I get this error then:

```
 Error
styfle/cancel-workflow-action@0.12.1, actions/checkout@v4, actions/cache@v4, jurplel/install-qt-action@v4, transifex/cli-action@v2, and 1 other action are not allowed to be used in T-X/MuseScore. Actions in this workflow must be: within a repository owned by T-X.
```
https://github.com/T-X/MuseScore/actions/runs/15619552378

cbjeukendrup commented:
It sounds like you must have some setting enabled in your repository, because I don't recognise that... maybe https://docs.github.com/en/repositories/managing-your-repositorys-settings-and-features/enabling-features-for-your-repository/managing-github-actions-settings-for-a-repository helps?

T-X committed:
Fix build errors with recent versions of cmake

This fixes error messages like the following:

```
CMake Error at CMakeLists.txt:1 (cmake_minimum_required):
-- Configuring incomplete, errors occurred!
  Compatibility with CMake < 3.5 has been removed from CMake.

  Update the VERSION argument <min> value.  Or, use the <min>...<max> syntax
  to tell CMake that the project requires at least <min> but has been updated
  to work with policies introduced by <max> or earlier.

  Or, add -DCMAKE_POLICY_VERSION_MINIMUM=3.5 to try configuring anyway.
```

Signed-off-by: Linus Lüssing <linus.luessing@c0d3.blue>

T-X committed:
Fix builds with pkg-config

The $HOME directory differs between the musescore_deps repository and
the MuseScore repository. The former uses
/Users/runner/work/musescore_deps/musescore_deps/ while the latter
uses /Users/runner/. This leads to a mismatch between the directories in
listed in a .pc pkg-config file created by this repository and the patch
where libraries from the musescore_deps tarball are actually placed
during a MuseScore MacOS CI build.

This does not seem to be an issue for the current libraries which all
also provide a .cmake file for a cmake find_package(). However the
libInstPatch we are going to install only provides a .pc file and no
.cmake file. Therefore this change makes sure that libraries will be
prepared in /Users/runner/musescore_deps_macos/ and not in
/Users/runner/work/musescore_deps/musescore_deps/musescore_deps_macos/.

Signed-off-by: Linus Lüssing <linus.luessing@c0d3.blue>

T-X committed:
Add Glib for libInstPatch, for DLS support

The libInstPatch we are going to add for DLS support requires the Glib
library. Unfortunately, homebrew does not seem to provide a universal,
combined Glib for both x86_64 and armhf. Therefore this will fetch
each variant via homebrew individually and will then merge them
through lipo.

Signed-off-by: Linus Lüssing <linus.luessing@c0d3.blue>

T-X commented:
Could now build things fine through the CI.

Changelog v2:
* added a fix for the cmake version
* added a fix for pkg-config builds
* adding glib via homebrew + lipo

Also MuseScore builds for me with this. I could not runtime test the resulting image though.

I could imagine that libglib-2.0.0.dylib, libgobject-2.0.0.dylib and libgthread-2.0.0.dylib might still need to be added to the musescore_deps_macos.tar.gz for MuseScore to start. As things seem to be built dynamically linked and not statically linked. Or the other (easier?) alternative might be to have lipo merge the .a instead of the .dylib files and then build libinstpatch with BUILD_SHARED_LIBS=OFF instead, maybe? But just speculation, as I said before, I don't have a Mac device to test this on at hand. And the CI does not complain anymore now.
